runtime/runtime/src/lib.rs

```diff
@@ -1820,7 +1822,7 @@ impl Runtime {
             proof,
             delayed_receipts_count: delayed_receipts.len(),
             metrics: Some(metrics),
-            congestion_info: congestion_info.map(|info| info.congestion_info),
+            congestion_info: congestion_control.map(|c| *c.congestion_info()),
```

Comment:
This fixes a bug introduced in #11307.
We must return the modified congestion info, not the input congestion info.

The tests added by in this PR catches it, while existing tests didn't.